### PR TITLE
10013 loan data export

### PR DESCRIPTION
**Pull request by DevneyHamilton**

### CODE REVIEW

File: app/models/accounting/transaction.rb [hunk 1]
```diff
@@ -84,6 +84,20 @@ class Accounting::Transaction < ApplicationRecord
     order(:txn_date, "options.position", :created_at)
   }
   scope :interest_type, -> { where(loan_transaction_type_value: LOAN_INTEREST_TYPE) }
```
Comment (DevneyHamilton) [inline app/models/accounting/transaction.rb]: should I replace usages of the  :interest_type scope with new :by_type('interest') scope?

**Comment by emitche** [comment]
Base branch can now be set to `develop` as branch `10012_export_to_csv` has been merged into develop.

File: app/models/accounting/transaction.rb (hunk 1 above)
Comment (meltheadorable) [reply app/models/accounting/transaction.rb]: You can replace the internal implementation so it's `scope :interest_type, -> { by_type(LOAN_INTEREST_TYPE) }`

File: app/models/accounting/transaction.rb [hunk 2]
```diff
@@ -80,10 +80,22 @@ class Accounting::Transaction < ApplicationRecord
 
   scope :standard_order, -> {
     joins("LEFT OUTER JOIN options ON options.option_set_id = #{loan_transaction_type_option_set.id}
-      AND options.value = accounting_transactions.loan_transaction_type_value").
-    order(:txn_date, "options.position", :created_at)
+      AND options.value = accounting_transactions.loan_transaction_type_value")
+      .order(:txn_date, "options.position", :created_at)
   }
   scope :interest_type, -> { where(loan_transaction_type_value: LOAN_INTEREST_TYPE) }
+  scope :by_type, lambda { |type| where(loan_transaction_type_value: type) }
```
Comment (meltheadorable) [inline app/models/accounting/transaction.rb]: You should be able to use `-> (type) { }` here and in the other one, which is style-preferred.

File: app/models/accounting/transaction.rb [hunk 3]
```diff
@@ -80,10 +80,22 @@ class Accounting::Transaction < ApplicationRecord
...
+  scope :in_date_range, lambda { |since_date, until_date|
```
Comment (meltheadorable) [inline app/models/accounting/transaction.rb]: `do ... end` syntax for multi-line

**Review by meltheadorable — APPROVED** [review]
Looks great, overall.

**Pull request merged**